pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -204,6 +204,19 @@ func (t *ResolvedPipelineRunTask) parentTasksSkip(facts *PipelineRunFacts) bool
 	return false
 }
 
+// IsFinallySkipped returns true if a finally task is not executed and skipped due to task result validation failure
+func (t *ResolvedPipelineRunTask) IsFinallySkipped(facts *PipelineRunFacts) bool {
```

Comment:
im surprised we need special "is skipped" logic for a finally task - wouldnt the logic for whether or not a task is skipped be the same as for non-finally tasks? (i.e. it depends on a task that failed or was skipped?)